### PR TITLE
core: Uploading an image with an existing disk id

### PR DESCRIPTION
**Pull request by ArtiomDivak**
While uploading users can choose disk-id for their upload and if
the disk-id was already exists the engine will make a new image
in lock state and fail the process. now it will check the disk-id before
creating and will fail if there is another disk-id with the same number.

Bug-Url: https://bugzilla.redhat.com/2095215
Signed-off-by: Artiom Divak <adivak@redhat.com>

### CODE REVIEW

**Comment by ahadas** [comment]
@ArtiomDivak I didn't look at the code changes but I've noticed that the commit here also doesn't point to the bug, please take a look at how it was done [here](https://github.com/oVirt/ovirt-engine/pull/495/commits/51f85f6efda9bb6a72b70429b2343224148ae057)

File: frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties [hunk 1]
```diff
@@ -289,6 +289,7 @@ ACTION_TYPE_FAILED_ILLEGAL_OS_TYPE_IS_NOT_SUPPORTED_BY_ARCHITECTURE_TYPE=Cannot
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected display type is not supported by the operating system.
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_FIRMWARE=Cannot ${action} ${type}. Selected display type is not supported by the firmware.
 ACTION_TYPE_FAILED_ILLEGAL_WATCHDOG_MODEL_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected watchdog model is not supported by the operating system.
+ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. One of the Template Disks already exists.
```
Comment (dupondje) [inline frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties:292]: Its not a Template disk. But just a disk.

` ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. One of the disks already exists.`

File: frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties [hunk 2]
```diff
@@ -289,6 +289,7 @@ ACTION_TYPE_FAILED_ILLEGAL_OS_TYPE_IS_NOT_SUPPORTED_BY_ARCHITECTURE_TYPE=Cannot
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected display type is not supported by the operating system.
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_FIRMWARE=Cannot ${action} ${type}. Selected display type is not supported by the firmware.
 ACTION_TYPE_FAILED_ILLEGAL_WATCHDOG_MODEL_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected watchdog model is not supported by the operating system.
+ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. The following disks already exist.
```
Comment (ahadas) [inline frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties:292]: and then let's list the disk aliases? :)

File: frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties [hunk 3]
```diff
@@ -289,6 +289,7 @@ ACTION_TYPE_FAILED_ILLEGAL_OS_TYPE_IS_NOT_SUPPORTED_BY_ARCHITECTURE_TYPE=Cannot
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected display type is not supported by the operating system.
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_FIRMWARE=Cannot ${action} ${type}. Selected display type is not supported by the firmware.
 ACTION_TYPE_FAILED_ILLEGAL_WATCHDOG_MODEL_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected watchdog model is not supported by the operating system.
+ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. The following disk already exist ${diskAlias}.
```
Comment (ahadas) [inline frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties:292]:
```suggestion
ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. The following disk already exist\:  ${diskAlias}. 
```

**Review by ahadas — COMMENTED** [review]
minor suggestion, besides that looks good

**Review by ahadas — APPROVED** [review]

File: frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties [hunk 4]
```diff
@@ -289,6 +289,7 @@ ACTION_TYPE_FAILED_ILLEGAL_OS_TYPE_IS_NOT_SUPPORTED_BY_ARCHITECTURE_TYPE=Cannot
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected display type is not supported by the operating system.
 ACTION_TYPE_FAILED_ILLEGAL_VM_DISPLAY_TYPE_IS_NOT_SUPPORTED_BY_FIRMWARE=Cannot ${action} ${type}. Selected display type is not supported by the firmware.
 ACTION_TYPE_FAILED_ILLEGAL_WATCHDOG_MODEL_IS_NOT_SUPPORTED_BY_OS=Cannot ${action} ${type}. Selected watchdog model is not supported by the operating system.
+ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS=Cannot ${action} ${type}. The following disk already exist\: ${diskAlias}.
```
Comment (ljelinkova) [inline frontend/webadmin/modules/frontend/src/main/resources/org/ovirt/engine/ui/frontend/AppErrors.properties:292]: If the user selects and ID that already exists, the message would be more clear like this: The selected ID is already used by the following existing disk: ${diskAlias}.

**Review by ahadas — APPROVED** [review]

**Review by bennyz — APPROVED** [review]

**Review by ljelinkova — CHANGES_REQUESTED** [review]
I think you forgot to add `ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS` to `AppErrors.java`

**Comment by ahadas** [comment]
> I think you forgot to add `ACTION_TYPE_FAILED_DISK_ALREADY_EXISTS` to `AppErrors.java`

indeed

**Review by ahadas — APPROVED** [review]

**Pull request merged**